contracts/MsgOracle.sol

```diff
@@ -0,0 +1,83 @@
+pragma solidity ^0.5.10;
+
+import "node_modules/openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+/**
+@title MsgOracle
+@author Rinke Hendriksen <rinke@ethswarm.org>
+@notice Set the honey prices for various messages in Swarm
+@dev This on-chain msgOracle must be queried by a swarm-node. A query for all msgPrices simultaneously should result in the following object:
+msgPrices: {TTL, prices: {swarmMsgX: [<validFrom0, price>, <validFromN, price>], swarmMsgX: [<validFrom0, price>, <validFromN, price>]}}.
+this object expires after TTL seconds. The prices object holds all message types with their respective price arrays.
+In the price array of a message, multiple prices can be quoted. The chosen price is the most-recent price.
+*/
+contract MsgOracle is Ownable {
+    /*
+    currentTTL is the TTL applied by nodes (how long is a query to this oracle valid)
+    oldTTL is used internally when lastUpdated is not longer than oldTTL ago,
+    furthermore, we don't allow updating TTL when lastUpdated is not currentTTL seconds ago.
+    */
+    uint256 public TTL;
+    uint256 private oldTTL;
+    uint256 private lastUpdated;
+
+    event LogNewTTL(uint256 TTL);
+    event LogSetMsgPrice(bytes32 indexed swarmMsg, uint256 price, uint256 validFrom);
+    event LogRevertMsgPrice(bytes32 indexed swarmMsg, uint256 price, uint256 validFrom);
+    /**
+    @notice Sets an initial value of TTL and sets the owner
+    @dev The owner can be a EOA or a smart-contract (with any arbitrary governance structure) which can call all functions from this contract
+    @param _TTL The initial TTL, effective immediately.
+    */
+    constructor(uint256 _TTL) public Ownable() {
+        TTL = _TTL;
+        lastUpdated = now;
+        emit LogNewTTL(TTL);
+    }
+
+    /**
+    @notice Sets a new value of TTL.
+    @dev The TTL is the duration in seconds of the validity of a query to this oracle.
```

Comment:
@ralph-pichler , what do you think about setting the unit of TTL to be in blocks?

Reply:
Not a fan of the idea. Block times are too unpredictable (e.g. an unchanged TTL would be longer the closer we are to the difficulty bomb).

Reply:
At the same time, relying on seconds will make it unpredictable when a node can expect an update to TTL and the time of the oracle-component (and the logic based on this time) of the swarm-node must be synchronized with the Ethereum block time at all time

Reply:
If TTL is given in blocks, the node can just count blocks and reason based on this. It is much simpler. TTL can always be set to fewer blocks, to compensate for the difficulty bomb

Reply:
If we count blocks anyway we can also look at their timestamp and make the decision based on that. Either way the network will have to be able to withstand the nodes being out of sync with prices for a short amount of time (That's unavoidable I'm afraid, we would either have to assume all nodes have perfectly synced clocks or receive blocks simultaneously, both scenarios are not realistic).